### PR TITLE
Pigweed RPC support

### PR DESCRIPTION
**Pull request by ATmobica**
#### Problem
Add Pigweed RPC support to Mbed CHIP integration

 #### Summary of Changes
Improve mbed logging - mbed trace usage
Add pigweed-app example
Add pw_rpc lib building to mbed config
Fix pigweed cmake version
Add mbed Pigweed logger and logger mutex
Add C++17 standard support
Remove time.h from BSD socket - use toolchain implementation
Add pigweed consoles to supported examples
Add RPC support for lighting app
Upload python_env as an artifact for pigweed apps
Add lighting-app and pigweed-app integration tests

### CODE REVIEW

**Review by lukgni — APPROVED** [review]
LGTM!
I only wonder which PR should be merged first: #177 or this one.
Both PRs put some changes into lighting-app example (but it shouldn't generate many conflicts).
Additionally we're improving mbed_trace support here by introducing platform/mbed/Logging module so after any of these two PR's will be merged, second one will have to be updated to reflected changes introduced by the first one.
Personally, I think better idea would be to merge  #177 first (if it will be ready), then update this PR and review it again but well, it's up to you guys.

**Comment by ATmobica** [comment]
@lukgni I agree with you. The first can be  #177 and moreover, it would be nice to merge changes from CI branch before. They contain the new structure of mbed test driver and here we add the new test cases. It would be better to update this PR with changes from CI and then merge this one.

**Review by tymoteuszblochmobica — APPROVED** [review]
LGTM

**Review by pan- — APPROVED** [review]
This require a rebase since #177 has been merged.

File: examples/common/pigweed/mbed/PigweedLoggerMutex.h
Comment (ATmobica) [reply examples/common/pigweed/mbed/PigweedLoggerMutex.h:32]: That's a good question for the CHIP maintainers. I implemented this in line with the general convention.

File: examples/lighting-app/efr32/README.md
Comment (ATmobica) [reply examples/lighting-app/efr32/README.md:14]: Some mistake, I revert it.

**Review by pan- — APPROVED** [review]

**Pull request merged**
